pkg/cinderapi/volumes.go

```diff
@@ -24,19 +26,46 @@ func GetVolumes(parentName string, name string, extraVol []cinderv1beta1.CinderE
 		},
 	}
 
-	return append(cinder.GetVolumes(parentName, false, extraVol, cinder.CinderAPIPropagation), backupVolumes...)
+	for _, secretName := range secretNames {
+		secretVol := corev1.Volume{
+			Name: secretName,
+			VolumeSource: corev1.VolumeSource{
+				Secret: &corev1.SecretVolumeSource{
+					SecretName:  secretName,
+					DefaultMode: &config0640AccessMode,
+				},
+			},
+		}
+
+		volumes = append(volumes, secretVol)
+	}
```

Comment:
It should be possible to do this is a common method to reuse in all the services, right?

Reply:
Yeah, I think so! I'll work on that.

Reply:
Deduped this in latest patchset